_posts/2016-09-04-build-maintainable-systems-with-hexagonal-architecture.md

```diff
@@ -81,7 +95,7 @@ Another secondary port is the notification service. If the player has written hi
 <script src="http://gist-it.appspot.com/http://github.com/iluwatar/java-design-patterns/raw/master/hexagonal/src/main/java/com/iluwatar/hexagonal/notifications/LotteryNotifications.java?slice=26:"></script>
 <script src="http://gist-it.appspot.com/http://github.com/iluwatar/java-design-patterns/raw/master/hexagonal/src/main/java/com/iluwatar/hexagonal/notifications/LotteryNotificationsImpl.java?slice=26:"></script>
 
-The methods in LotteryNotificationsImpl adapter are simple System.out printers so the implementation is trivial to understand.
+The methods in `LotteryNotificationsImpl` adapter are simple `System.out` printers so the implementation is trivial to understand.
```

Comment:
A better name for `LotteryNotificationsImpl` would be `ConsoleNotifications`. Also if the interface represents an entity that will send out notifications then the name can be something like `LotteryEventNotifier`, and the console implementation can have name `ConsoleNotifier`.
Also in the code there are `*Impl` classes, I am not fond of names like `*Impl`, rather the implementation should describe how it does, what it does. Plus why do classes like `LotterySystemImpl` hardcode their dependency on `LotteryNotificationsImpl`, `LotteryTicketInMemoryRepository`, `WireTransfersImpl`. Out of these lottery ticket repository and Wire Transfers are dependencies of the system and should be injected using constructor. What if we wanted to test the system using `MongoTicketRepository`?